### PR TITLE
revenant cleanup

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR 
replaces all the old revenant store logic with the new storesystem

holy shit dat cleanup :eyes:

**Screenshots**

**Changelog**

no cl no fun

### CODE REVIEW

File: Content.Server/Store/Systems/StoreSystem.Ui.cs [hunk 1]
```diff
@@ -61,14 +61,15 @@ public void UpdateUserInterface(EntityUid? user, StoreComponent component, Bound
             ui = component.Owner.GetUIOrNull(StoreUiKey.Key);
             if (ui == null)
             {
-                Logger.Error("No Ui key.");
```
Comment (mirrorcult) [inline Content.Server/Store/Systems/StoreSystem.Ui.cs:64]: this should be removed or use an actual sawmill with more info

File: Resources/Prototypes/Catalog/revenant_catalog.yml [hunk 2]
```diff
@@ -1,32 +1,51 @@
...
-  price: 125
\ No newline at end of file
+  productAction: RevenantMalfunction
+  cost: 
+    StolenEssence: 125
+  categories:
+  - RevenantAbilities
+  conditions:
+  - !type:ListingLimitedStockCondition
+    stock: 1
```
Comment (mirrorcult) [inline Resources/Prototypes/Catalog/revenant_catalog.yml:51]: are these intended to have limited stock? it says 'per use' which implies multiple times?

also why does it say 'costs X essence' but then it actually costs a different amount

**Review by mirrorcult — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Catalog/revenant_catalog.yml (hunk 2 above)
Comment (EmoGarbage404) [reply Resources/Prototypes/Catalog/revenant_catalog.yml:51]: This is a symptom of me being completely incapable of describing things well. The idea is that it's a one-time unlock cost in the menu and then each menu item has an independent use cost each time it's triggered. I keep getting asked about this so i'll remove that portion of it.

Comment (mirrorcult) [reply Resources/Prototypes/Catalog/revenant_catalog.yml]: okay that makes sense

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**